### PR TITLE
WT-4580 Make transactions with ignore_prepare=true readonly.

### PR DESCRIPTION
**Pull request by michaelcahill**

### CODE REVIEW

**Comment by agorrod** [comment]
Closing for now, since this is blocked on some MongoDB server changes. Will resurrect when those are finalized.

**Comment by luke-pearson** [comment]
Re-opening this PR as the related server work is now complete.

**Comment by luke-pearson** [comment]
retest this please

File: src/include/txn.i [hunk 1]
```diff
@@ -511,9 +511,14 @@ __wt_txn_modify(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 
 	txn = &session->txn;
 
-	if (F_ISSET(txn, WT_TXN_READONLY))
+	if (F_ISSET(txn, WT_TXN_READONLY)) {
+		if (F_ISSET(txn, WT_TXN_IGNORE_PREPARE))
```
Comment (louiswilliams) [inline src/include/txn.i]: @luke-pearson It looks like this returns  ENOTSUP even when `ignore_prepare=force`. Is there more work needed to suppress the error in that case? I applied this locally with the server changes, but I still get failures.

File: src/txn/txn.c [hunk 2]
```diff
@@ -529,9 +529,14 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
...
-	    (cval.len > 0 && WT_STRING_MATCH("force", cval.str, cval.len)))
-		F_SET(txn, WT_TXN_IGNORE_PREPARE);
+	if (cval.val)
+	{
+		if (cval.len > 0 &&
+		    WT_STRING_MATCH("force", cval.str, cval.len))
+			F_SET(txn, WT_TXN_IGNORE_PREPARE);
+		else
+			F_SET(txn, WT_TXN_IGNORE_PREPARE | WT_TXN_READONLY);
+	}
```
Comment (michaelcahill) [inline src/txn/txn.c]: Suggest losing the top `if (cval.val)` and replacing with `if (cval.len > 0 && WT_STRING_MATCH("force", cval.str, cval.len) ... else if (cval.val) ...`

File: src/include/txn.i (hunk 1 above)
Comment (luke-pearson) [reply src/include/txn.i]: I've put in a fix for the issue, could you please retest?

Comment (louiswilliams) [reply src/include/txn.i]: This does work now, thanks.

**Pull request merged**